src/vcpkg/tools.cpp

```diff
@@ -585,35 +585,56 @@ gsutil version: 4.58
                 // First deal with specially handled tools.
                 // For these we may look in locations like Program Files, the PATH etc as well as the auto-downloaded
                 // location.
+                auto& fs = paths.get_filesystem();
                 if (tool == Tools::CMAKE)
                 {
                     if (get_environment_variable("VCPKG_FORCE_SYSTEM_BINARIES").has_value())
                     {
-                        return {"cmake", "0"};
+                        auto tool_path = fs.find_from_PATH("cmake");
+                        if (tool_path.empty())
+                        {
+                            Checks::exit_with_message(VCPKG_LINE_INFO, "Error: Couldn't find %s form your path", tool);
```

Comment:
It would be better if this block duplicated 5 times was a function call.

Reply:
```suggestion
                            Checks::exit_with_message(VCPKG_LINE_INFO, "Error: Couldn't find %s on the PATH", tool);
```

Reply:
Updated! thanks for your review!